### DISCUSSION TITLE
Should I create an Architecture.md ?

### DISCUSSION THREAD

**Discussion by erezsh** [lark-parser/lark#832] [General]
After reading this blogpost : 
https://matklad.github.io//2021/02/06/ARCHITECTURE.md.html

I'm wondering if I should write an architecture file for Lark, explaining shortly what are the main modules and how they relate to each other.

If enough people are interested I will do it.

**Comment by MegaIng** [comment]
I would say this is a good idea. I already added info for something like this for [new parameters](https://github.com/lark-parser/lark/blob/ace1c9f4ea9b2c5217d00e74bbdd3c24d1e394b6/lark/lark.py#L108) at some point because of how split up the system was (not that we can really change that).

**Comment by ThatXliner** [comment]
I think this is a good idea especially for people wanting to port Lark.
